Source/DafnyTestGeneration/DafnyInfo.cs

```diff
@@ -105,32 +470,116 @@ private class DafnyInfoExtractor : BottomUpVisitor {
       }
 
       private new void Visit(Method m) {
-        var methodName = m.Name;
-        if (path.Count != 0) {
-          methodName = $"{string.Join(".", path)}.{methodName}";
-        }
-        if (m.HasStaticKeyword || !insideAClass) {
-          info.isStatic.Add(methodName);
-        }
-
-        info.returnTypes[methodName] =
-          m.Outs.Select(arg => arg.Type.ToString()).ToList();
-        info.formalsTypes[methodName] = m.Ins.Select(arg => arg.Type).ToList();
+        info.methods[m.FullDafnyName] = m;
       }
 
       private new void Visit(Function f) {
-        var methodName = f.Name;
-        if (path.Count != 0) {
-          methodName = $"{string.Join(".", path)}.{methodName}";
+        info.functions[f.FullDafnyName] = f;
+      }
+    }
+
+    private class ClonerWithSubstitution : Cloner {
```

Comment:
Does `Substituter` not work for this use case?

Reply:
The difference with `Substituter` is, I think, that I need to work with a resolved AST.